examples/nlp/lasertagger/lasertagger_main.py

```diff
@@ -0,0 +1,406 @@
+    """Calculates the number of steps.
+
+  Args:
+	num_examples: Number of examples in the dataset.
+	batch_size: Batch size.
+	num_epochs: How many times we should go through the dataset.
+	warmup_proportion: Proportion of warmup steps.
+
+  Returns:
+	Tuple (number of steps, number of warmup steps).
+  """
+    steps = int(num_examples / batch_size * num_epochs)
+    warmup_steps = int(warmup_proportion * steps)
+    return steps, warmup_steps
+
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    if args.command == 'train':
+        train_examples, num_train_examples = torch.load(args.train_file_preprocessed)
+        if args.eval_file_preprocessed:
+            eval_examples, num_eval_examples, eval_special_tokens = torch.load(args.eval_file_preprocessed)
+    test_examples, num_test_examples, test_special_tokens = torch.load(args.test_file_preprocessed)
+
+    label_map = utils.read_label_map(args.label_map_file)
+    num_tags = len(label_map)
+
+    config = None
+    with open(args.model_config_file, "r", encoding="utf-8") as reader:
+        text = reader.read()
+        config = json.loads(text)
+
+    nf = nemo.core.NeuralModuleFactory(
+        local_rank=args.local_rank,
+        optimization_level=args.amp_opt_level,
+        log_dir=args.work_dir,
+        create_tb_writer=True,
+        files_to_copy=[__file__],
+        add_time_to_log_dir=False,
+    )
+
+    encoder = nemo_nlp.nm.trainables.huggingface.BERT(pretrained_model_name=args.pretrained_model_name)
+
+    tokenizer = nemo_nlp.data.NemoBertTokenizer(pretrained_model=args.pretrained_model_name)
+
+    # tokenizer.add_special_tokens({"additional_special_tokens": test_special_tokens})
+    # encoder.resize_token_embeddings(len(tokenizer))
+
+    # vocab_size = 8 * math.ceil(tokenizer.vocab_size / 8)
+    # tokens_to_add = vocab_size - tokenizer.vocab_size
+
+    encoder = nemo_nlp.nm.trainables.huggingface.BERT(pretrained_model_name=args.pretrained_model_name)
+
+    hidden_size = encoder.hidden_size
+
+    # device = encoder.bert.embeddings.word_embeddings.weight.get_device()
+    # zeros = torch.zeros((tokens_to_add, config['decoder_hidden_size'])).to(device=device)
+    # # encoder.bert.embeddings.word_embeddings.weight.data = torch.cat(
+    # #     (encoder.bert.embeddings.word_embeddings.weight.data, zeros)
+    # # )
+
+    # Size of the output vocabulary which contains the tags + begin and end
+    # tokens used by the Transformer decoder.
+    output_vocab_size = num_tags
+
+    decoder = nemo_nlp.nm.trainables.TransformerDecoderNM(
+        d_model=config['decoder_hidden_size'],
+        d_inner=config['decoder_filter_size'],
+        num_layers=config['decoder_num_hidden_layers'],
+        num_attn_heads=config['decoder_num_attention_heads'],
+        ffn_dropout=0.1,
+        vocab_size=output_vocab_size,
+        attn_score_dropout=0.1,
+        max_seq_length=args.max_seq_length,
+        embedding_dropout=0.1,
+        hidden_act='relu',
```

Comment:
why parameters are partially hardcoded here, and partially in config file?

Reply:
actually that comment goes to the following modules too...

Reply:
ok will add, argparse based parameter management.

Reply:
so 'import_from_config()' has 'overwrite_params` that enables the user to inject values that are coming from argparse: 
https://github.com/NVIDIA/NeMo/blob/e095496e1f24c60eeef0c6920af5a7fceb1f06e0/nemo/core/neural_modules.py#L326